src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -407,8 +436,54 @@ static URLConnection openURLConnection(URL url, String credentials, String proxy
             String encoding = Base64.encode(proxyCredentials.getBytes("UTF-8"));
             con.setRequestProperty("Proxy-Authorization", "Basic " + encoding);
         }
-        if (con instanceof HttpsURLConnection && sslSocketFactory != null) {
+
+        if (insecure && con instanceof HttpsURLConnection) {
+            System.out.println(String.format("Insecure Status: %s", insecure));
```

Comment:
Better to use a logger instead of System.out.println.

Reply:
It requires a wider refactoring, the option is being used in other places